Essentials/src/com/earth2me/essentials/commands/Commandmore.java

```diff
@@ -17,22 +18,42 @@ public Commandmore() {
     @Override
     public void run(final Server server, final User user, final String commandLabel, final String[] args) throws Exception {
         final ItemStack stack = user.getItemInHand();
-        if (stack == null) {
+        if (stack == null || stack.getType() == Material.AIR) {
             throw new Exception(tl("cantSpawnItem", "Air"));
         }
 
-        if (stack.getAmount() >= ((user.isAuthorized("essentials.oversizedstacks")) ? ess.getSettings().getOversizedStackSize() : stack.getMaxStackSize())) {
+        boolean canOversized = user.isAuthorized("essentials.oversizedstacks");
+        if (stack.getAmount() >= ((canOversized) ? ess.getSettings().getOversizedStackSize() : stack.getMaxStackSize())) {
             throw new Exception(tl("fullStack"));
         }
+
         final String itemname = stack.getType().toString().toLowerCase(Locale.ENGLISH).replace("_", "");
         if (!user.canSpawnItem(stack.getType())) {
             throw new Exception(tl("cantSpawnItem", itemname));
         }
-        if (user.isAuthorized("essentials.oversizedstacks")) {
-            stack.setAmount(ess.getSettings().getOversizedStackSize());
+
+        int newStackSize = stack.getAmount();
+        if (args.length >= 1) {
+            try {
+                int increment = Integer.parseInt(args[0]);
+                if (increment < 1) {
+                    throw new NumberFormatException("increment is less than or equal to zero");
+                }
+                newStackSize += increment;
+            } catch(NumberFormatException e) {
+                throw new Exception(tl("nonZeroPosNumber"));
+            }
```

Comment:
Not really a big fan of this kind of try/throw/catch, looks very messy. Maybe use `NumberUtil::isInt(String)` here instead, and additionally check that the parsed number is greater than zero. If either fails, throw the exception for `nonZeroPosNumber`.